en/debug/jtag_pins.md

```diff
@@ -0,0 +1,130 @@
+The physical connector and pin order for this port varies across Pixhawk FMU versions.
+
+> **Note** The **Pixhawk Debug Port** also exposes a UART for access to the [System Console](../debug/system_console.md) (a console that provides low-level access to the system, debug output and analysis of the system boot process).
+
+<!--
+### FMUv2
+
+Does this have a debug port? where is spec
+Is it a Hirose DF13
+
+
+### FMUv3
+
+Does this have a debug port? where is spec
+Is it a Hirose DF13
+-->
+
+### FMUv4/FMUv5
+
+FMU V4,5 has a 6 pin JST GH
+
+[Pixracer](http://docs.px4.io/master/en/flight_controller/pixracer.html#pinouts): [Pixhawk debug port](https://pixhawk.org/pixhawk-connector-standard/#dronecode_debug) (JST SM06B connector)
+
+Pin | Signal | Volt
+--- | --- | ---
+1 (red) | VCC TARGET SHIFT | +3.3V
+2 (blk) | CONSOLE TX (OUT) | +3.3V
+3 (blk) | CONSOLE RX (IN) | +3.3V
+4 (blk) | SWDIO | +3.3V
+5 (blk) | SWCLK | +3.3V
+6 (blk) | GND | GND
+
+
+### FMUv5X
+
+FMUv5X will have a [10-pin JST SUR](https://order.jst-mfg.com/InternetShop/app/index.php?back=2&product=53574000&jgcd=0#showProductDetail).
+It adds JTAG pins: SWO, nRST. TRACECLK, TRACED0 Pins 7 and 8 are GPIOS to gate timing or 1 wire trace (full trace is on).
+<!-- is this link/example correct type for connector? -->
+
+FMU Debug Port Pinout
+
+Pin | Signal | Volt
+--- | --- | ---
+1 (red) | FMU VDD | +3.3V
+2 (blk) | CONSOLE TX (OUT) | +3.3V
+3 (blk) | CONSOLE RX (IN) | +3.3V
+4 (blk) | SWDIO | +3.3V
+5 (blk) | SWCLK | +3.3V
+6 (blk) | SWO | +3.3V
+7 (blk) | TRACECLK | +3.3V
+8 (blk) | TRACED0 | +3.3V
+9 (blk) | nRST | +3.3V
+10 (blk) | GND | GND
+
+
+> **Note** This board will have an additional 8-pin JST SUR for ARM ETM TRACE).
+
+
+## Hex Cube
+
+Cube debug port information can be found here: [Debug ports](http://docs.px4.io/master/en/flight_controller/pixhawk-2.html#debug-ports).
+
+If pulled apart the Cube has a 6 pin JST SUR connector (marked as J10) for both FMU and IO debugging.
+
+<!--
+Does the numbering match all the other 6 pin port examples.
+What label is the IO port marked as?
+So obviously it is much easier to just connect to the carrier board port. Who would know what interface that is. PhilipR?
+-->
+
+
+## Kakute F7
+
+There are Test points on the board.
+<!-- I take that to mean that there are SWDIO etc pins somewhere. How do I find out where? -->
+
+UART3 RX and TX are configured for System console. 
+
+
+## Hovergames RDDRONE-FMUK66
+
+The Hovergames RDDRONE-FMUK66 has a similar debug interface to Pixhawk (it supports an additional JTAG pin - nRST) and uses a JST-GH connector). 
+The debug interface is documented here: [RDDRONE-FMUK66 Debug interface](https://nxp.gitbook.io/hovergames/rddrone-fmuk66/connectors/debug-interface-dcd-lz)
+
+![Debug RDDRONE-FMUK66 using Jlink and debug interface](../../assets/debug/jlink_hovergames.png)
```

Comment:
@davids5 This is your diagram. What is the orange thing in the middle and where does that connection go? 

Is this just confusing? ie could they just wire up the lines directly?

Reply:
Debugging means you have a problem. Making you own wiring adds a whole set of  potential faults in your fault tree.  == MORE PROBLEMS. Therefore debugging equipment needs to be rock solid and stable.  NXP, PX4, NscDG recognized this made adapter boards. 

![image](https://user-images.githubusercontent.com/1945821/68938668-adeea680-0753-11ea-9e48-520137f08ef4.png)

NXP
![image](https://user-images.githubusercontent.com/1945821/68940324-b0eb9600-0757-11ea-8326-05ef0f3aa1e7.png)

Dronecode 
![image](https://user-images.githubusercontent.com/1945821/68940262-88639c00-0757-11ea-8065-9807c5fbda28.png)

NscDG 
![image](https://user-images.githubusercontent.com/1945821/68940164-4f2b2c00-0757-11ea-9b6b-de89113088ea.png)

Reply:
The roll your own header uses cables like 
![image](https://user-images.githubusercontent.com/1945821/68941096-bc3fc100-0759-11ea-9614-74852d19f469.png)
 for devices that have special connectors or even just PAD